cmd/infra/azure/create.go

```diff
@@ -849,6 +853,9 @@ func assignServicePrincipalRoles(subscriptionID, managedResourceGroupName, nsgRe
 		role = "Azure Red Hat OpenShift Image Registry Operator Role"
 	case nodePoolMgmt:
 		scopes = append(scopes, vnetRG)
+		if assignCustomHCPRoles {
+			role = "Azure Red Hat OpenShift Cluster API Role"
```

Comment:
Wondering if this should be Azure Red Hat OpenShift NodePool Management Role to keep it agnostic from CAPI/CAPZ? That would be in line with what we do for AWS as well - https://github.com/openshift/hypershift/blob/103d1486a1326152865fc7dfb1e91d2cc147ef18/api/hypershift/v1beta1/aws.go#L508.

ref - https://github.com/openshift/hypershift/pull/4811#discussion_r1800604551